### ISSUE TITLE
Unable to get floating ip pools

### ISSUE THREAD

**Issue by ssibahi** [jenkinsci/openstack-cloud-plugin#335]
Jenkins 2.308 over docker
Openstack-cloud-plugin version tried 2.51-2.60

please note that over ver 2.51 i have managed to get the pool list but apply/save buttons are broken because of old version of the plugin.

The error causing the provisioned instance to spin up but without floating ip

![WhatsApp Image 2021-11-14 at 11 59 21 (1)](https://user-images.githubusercontent.com/36002056/141676463-96d9bfd4-d0a4-4259-b12d-fd7d2d536ce4.jpeg)
![111](https://user-images.githubusercontent.com/36002056/141676562-4d008377-4510-405b-b1f3-05d4bfe384c2.jpeg)

**Comment by olivergondza** [comment]
Thanks for the report!

What exact version of the plugin produced this specific stacktrace? Also, what are your networks and routers details? I speculate you might be running into this as some of your routers are not up, or connected to an external gateway. If confirmed this is the case, this will be reflected in the plugin.

**Comment by ssibahi** [comment]
@olivergondza,
It happened at ver 2.60
there is no problem with the network - also, ver 2.51 did managed to view the float ip pools.

**Comment by olivergondza** [comment]
@ssibahi, between 2.51 and 2.52 the plugin have switched from deprecated nova networking to neutron networking. For some reason, new code does not identify your pools correctly. However, you will have to provide the details of what your network and router details are to get this fixed.

Please attach result of this [diagnosing script](https://github.com/jenkinsci/openstack-cloud-plugin/blob/master/README.md#access-openstack-client-from-groovy-console):
```groovy
Jenkins.instance.clouds.each {
  if (it instanceof jenkins.plugins.openstack.compute.JCloudsCloud) {
    client = it.openstack.@clientProvider.get()
    println "Routers: " + client.networking().router().list()
    println "Networks: " + client.networking().network().list()
  }
}
```

**Comment by olivergondza** [comment]
Confirmed by other user, that this happens when there are routers without `external_gateway_info`. So in https://github.com/jenkinsci/openstack-cloud-plugin/blob/parent-2.60/plugin/src/main/java/jenkins/plugins/openstack/compute/internal/Openstack.java#L332, `r.getExternalGatewayInfo()` is `null`.

**Comment by olivergondza** [comment]
Released as 2.61
